src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -314,6 +315,18 @@ calculatePrincipalExchangePaymentClaims paymentScheduleAligned issuerPaysLeg cur
     in
       prepareAndTagClaims dateToDateClockTime principalClaims "Principal exchange payment"
 
+-- | Apply rounding convention to the rate used in a calculation period.
+-- Takes a Rounding FpML object as an input:
+-- https://www.fpml.org/spec/fpml-5-11-3-lcwd-1/html/confirmation/schemaDocumentation/schemas/fpml-shared-5-11_xsd/complexTypes/FloatingRateCalculation/finalRateRounding.html
+roundRate : BigNumeric -> Rounding -> Decimal
```

Comment:
Are we using `BigNumeric` because the rounding function are not defined for a `Decimal` (aka a `Numeric 10`)?

I think a sensible signature for this function would be `Decimal -> Rounding -> Decimal`, otherwise you will need to convert explicitly in your code as currently I assume you are using `Decimal` in the Fpml swap types

Reply:
I think you can actually use `ceiling`, `floor`, `round` defined in the `Prelude` without resorting to `BigNumeric`

Reply:
https://discuss.daml.com/t/reducing-precision-of-numeric-values/3080

Reply:
If, on the other hand, you are actually storing `BigDecimal`s, then we need to change contingent claims to work for a generic numeric type (currently it's hardcoded to `Decimal`)

Reply:
I don't really need `BigNumeric`, but I need rounding where I can specify both rounding direction and precision (number of digits).

Reply:
> Are we using `BigNumeric` because the rounding function are not defined for a `Decimal` (aka a `Numeric 10`)?

Yes, all the rounding functionality I needed worked out of the box there.

> I think a sensible signature for this function would be `Decimal -> Rounding -> Decimal`, otherwise you will need to convert explicitly in your code as currently I assume you are using `Decimal` in the Fpml swap types

I have changed the signature now.